plugins/labhub.py

```diff
@@ -242,14 +242,63 @@ def mark_cmd(self, msg, match):
                             bot_prefix=self.bot_config.BOT_PREFIX)
                         )
 
-    @re_botcmd(pattern=r'^assign\s+https://(github|gitlab)\.com/([^/]+)/([^/]+/)+issues/(\d+)',  # Ignore LineLengthBear, PyCodeStyleBear
-               re_cmd_name_help='assign <complete-issue-URL>',
-               flags=re.IGNORECASE)
-    def assign_cmd(self, msg, match):
+    @arg_botcmd('issue_reference', type=str)
+    def assign(self, msg, issue_reference):
         """Assign to GitLab and GitHub issues."""  # Ignore QuotesBear
-        org = match.group(2)
-        repo_name = match.group(3)[:-1]
-        iss_number = match.group(4)
+
+        # Complete URL to issue
+        def process_full_url(issue_reference):
+            rgx = r'https://(github|gitlab)\.com/([^/]+)/([^/]+/)+issues/(\d+)'
+            m = re.fullmatch(rgx, issue_reference, re.IGNORECASE)
+
+            if m is None:
+                return None
+
+            return m.group(2), m.group(3)[:-1], m.group(4)
+
+        # Short issue reference (e.g. `coala/corobo#12`)
+        def process_short_ref(issue_reference):
+            rgx = r'(.+?)/(.+?)#(\d+)'
+            m = re.fullmatch(rgx, issue_reference, re.IGNORECASE)
+
+            if m is None:
+                return None
+
+            return m.group(1), m.group(2), m.group(3)
+
+        # Super short issue reference (e.g. `#1234`)
+        def process_super_short_ref(issue_reference):
+            issue_rgx = r'#(\d+)'
+            issue_reference_match = re.fullmatch(issue_rgx, issue_reference)
+
+            if issue_reference_match is None:
+                return None
+
+            roomname_rgx = r'(.+?)/(.+)'
+            roomname_match = re.fullmatch(
+                roomname_rgx, msg.frm.room.idd, re.IGNORECASE)
+
+            if roommatch is None:
+                return None
+
+            return roommatch.group(1), roommatch.group(2), m.group(1)
```

Comment:
Same as above

Reply:
There is no `m` in this scope :/

Reply:
ah yeah this has to be the very first match variable from above, will correct it :3
(you already see that my tests aborted above because accessing the room name raised an exception :D)